src/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -9,13 +9,17 @@ namespace System.Threading
 {
     public sealed class Lock
     {
+        // The following constants define characteristics of spinning logic in the Lock class
+        const int SpinningNotInitialized = 0;
+        const int SpinningDisabled = -1;
+        const int MaxSpinningValue = 10000;
```

Comment:
I guess I am in the other camp then. :) I strongly prefer not to see numbers embedded into code and that constants with meaningful/self-explanatory names should be used instead. This is for various reasons starting from that a number does not describe its meaning; only one location to change if the value needs to be updated; making the code easier to read. For example, I think it is easier to grasp this line

```s_maxSpinCount = (Environment.ProcessorCount > 1) ? MaxSpinningValue : SpinningDisabled;```

than trying to decipher what the intention of ”-1” is in

```s_maxSpinCount = (Environment.ProcessorCount > 1) ? 10000 : -1;```

At least, that’s my preference. :)